test/files/neg/outer-ref-checks.check

```diff
@@ -16,9 +16,12 @@ outer-ref-checks.scala:43: warning: The outer reference in this type test cannot
 outer-ref-checks.scala:48: warning: The outer reference in this type test cannot be checked at run time.
     case _: Inner => // unchecked warning
           ^
+outer-ref-checks.scala:53: warning: The outer reference in this type test cannot be checked at run time.
+    case _: (Inner @unchecked) => // warning suppressed
+          ^
```

Comment:
`@unchecked` should suppress the warning here, no?

Reply:
I can be convinced so, but I wasn't 100% sure.  Does `@unchecked` just mean "don't give any warning"?  I thought it was only for some subset of warnings.  I'll look to fix this.

Reply:
I don't know about a "spec" for `@unchecked`, and I think it became what it is today organically...

This commit message is relevant: https://github.com/scala/scala/commit/e7e1c77092. Would be good to extend the scaladocs of unchecked.